src/libcore/slice.rs

```diff
@@ -1459,12 +1459,30 @@ pub mod bytes {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A, B> PartialEq<[B]> for [A] where A: PartialEq<B> {
     fn eq(&self, other: &[B]) -> bool {
-        self.len() == other.len() &&
-            order::eq(self.iter(), other.iter())
+        if self.len() != other.len() {
+            return false;
+        }
+
+        for i in 0..self.len() {
+            if !self[i].eq(&other[i]) {
+                return false;
+            }
+        }
+
+        true
     }
     fn ne(&self, other: &[B]) -> bool {
```

Comment:
The documentation for `PartialEq` states: "Any manual implementation of `ne` must respect the rule that `eq` is a strict inverse of `ne`; that is, !(a == b) if and only if a != b."
Under this assumption, `ne` can always re-use the implementation from the trait definition. I would expect the compiler to do a good job optimising it as it can fold the negation either on the possible outcomes or on the `ne` usage. Unless some benchmarking suggests otherwise, I would assume that avoiding the duplication is preferred.